go/common/viewingkey/viewing_key.go

```diff
@@ -308,49 +360,27 @@ func GenerateAuthenticationEIP712RawDataOptions(userID string, chainID int64) ([
 		Message:     message,
 	}
 
-	rawDataOptions := make([][]byte, 0)
 	rawData, err := getBytesFromTypedData(newTypeElement)
 	if err != nil {
 		return nil, err
 	}
-	rawDataOptions = append(rawDataOptions, rawData)
-
-	return rawDataOptions, nil
-}
-
-// CalculateUserIDHex CalculateUserID calculates userID from a public key
-// (we truncate it, because we want it to have length 20) and encode to hex strings
-func CalculateUserIDHex(publicKeyBytes []byte) string {
-	return hex.EncodeToString(CalculateUserID(publicKeyBytes))
-}
-
-// CalculateUserID calculates userID from a public key (we truncate it, because we want it to have length 20)
-func CalculateUserID(publicKeyBytes []byte) []byte {
-	return crypto.Keccak256Hash(publicKeyBytes).Bytes()[:20]
-}
 
-// CheckSignatureAndReturnAccountAddress checks if the signature is valid for hash of the message and checks if
-// signer is an address provided to the function.
-// It returns an address if the signature is valid and nil otherwise
-func CheckSignatureAndReturnAccountAddress(hashBytes []byte, signature []byte) (*gethcommon.Address, error) {
-	pubKeyBytes, err := crypto.Ecrecover(hashBytes, signature)
+	// add the JSON message to the list of messages
+	jsonData, err := json.Marshal(newTypeElement)
 	if err != nil {
 		return nil, err
 	}
 
-	pubKey, err := crypto.UnmarshalPubkey(pubKeyBytes)
-	if err != nil {
-		return nil, err
+	if hash {
+		return crypto.Keccak256(rawData), nil
 	}
+	return jsonData, nil
+}
 
-	r := new(big.Int).SetBytes(signature[:32])
-	s := new(big.Int).SetBytes(signature[32:64])
-
-	// Verify the signature and return the result (all the checks above passed)
-	isSigValid := ecdsa.Verify(pubKey, hashBytes, r, s)
-	if isSigValid {
-		address := crypto.PubkeyToAddress(*pubKey)
-		return &address, nil
+func GenerateMessage(encryptionToken string, chainID int64, version int, signatureType SignatureType, hash bool) ([]byte, error) {
```

Comment:
needs a doc. i can't figure out what "hash" is doing

Reply:
Added a comment.
The with the last parameter se to true we can get hashed version of the message (so the prefixes for personal sign is already taken care of, etc..). I noticed we used to get message and then hash it in multiple places in our code and decided to simplify it a bit and do it on only one place.

Reply:
got it, but I don't think it's a good idea because it mixes responsibilities.
Let's keep the method to always return the message - basically keep that its single responsibility.
And when you need hashes, you call the method and then hash explicitly

Reply:
Good point. It has a single responsibility + create a new function to create hash for each message